cmd/siftest/main.go

```diff
@@ -0,0 +1,175 @@
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			err = Export(clientCtx, viper.GetInt64("height"), viper.GetString("from"), viper.GetString("pool"))
+			if err != nil {
+				panic(err)
+			}
+
+			return nil
+		},
+	}
+	cmd.Flags().Int64("height", 0, "height at which to export")
+	cmd.Flags().String("pool", "", "pool to export")
+	cmd.Flags().String("from", "", "account to export")
+	flags.AddQueryFlagsToCmd(cmd)
+	_ = cmd.MarkFlagRequired("height")
+	_ = cmd.MarkFlagRequired("pool")
+	_ = cmd.MarkFlagRequired("from")
+	return cmd
+}
+*/
+
+func GetTestCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "test",
+		Short: "Run tests",
+		RunE:  runTest,
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.PersistentFlags().String(flags.FlagChainID, "", "The network chain ID")
+	return cmd
+}
+
+func GetVerifyCmd() *cobra.Command {
+	verifyCmd := &cobra.Command{
+		Use:   "verify",
+		Short: "Verify transaction results",
+	}
+
+	verifyCmd.AddCommand(GetVerifyRemove(), GetVerifyAdd(), GetVerifyOpen(), GetVerifyClose())
+
+	return verifyCmd
+}
+
+/* VerifySwap verifies amounts sent and received from wallet address.
+ */
+func VerifySwap(clientCtx client.Context, key keyring.Info) {
+
+}
+
+func GetVerifyOpen() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "open-position --height --from --collateralAmount --leverage --collateral-asset --borrow-asset",
+		Short: "Verify a margin long position open",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fmt.Printf("verifying open...\n")
+			clientCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			collateralAmount := sdk.NewUintFromString(viper.GetString("collateralAmount"))
+			leverageDec, err := sdk.NewDecFromStr(viper.GetString("leverage"))
+			if err != nil {
+				panic(err)
+			}
+
+			err = VerifyOpenLong(clientCtx,
+				viper.GetString("from"),
+				int64(viper.GetUint64("height")),
+				collateralAmount,
+				viper.GetString("collateral-asset"),
+				viper.GetString("borrow-asset"),
+				leverageDec)
+			if err != nil {
+				panic(err)
+			}
+
+			return nil
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+	//cmd.Flags().Uint64("height", 0, "height of transaction")
+	cmd.Flags().String("from", "", "address of transactor")
+	cmd.Flags().String("collateralAmount", "0", "collateral provided")
+	cmd.Flags().String("leverage", "0", "leverage")
+	cmd.Flags().String("collateral-asset", "", "collateral asset")
+	cmd.Flags().String("borrow-asset", "", "borrow asset")
+	_ = cmd.MarkFlagRequired("from")
+	_ = cmd.MarkFlagRequired("collateralAmount")
+	_ = cmd.MarkFlagRequired("leverage")
+	_ = cmd.MarkFlagRequired("collateral-asset")
+	_ = cmd.MarkFlagRequired("height")
+	return cmd
+}
+
+func VerifyOpenLong(clientCtx client.Context,
+	from string,
+	height int64,
+	collateralAmount sdk.Uint,
+	collateralAsset,
+	borrowAsset string,
+	leverage sdk.Dec) error {
+
+	return nil
+}
```

Comment:
is it actually needed if it returns `nil` ?

Reply:
keeping this here as WIP